browser/search_engine_provider_util.cc

```diff
@@ -27,18 +29,29 @@ void ToggleUseAlternativeSearchEngineProvider(Profile* profile) {
 void RegisterAlternativeSearchEngineProviderProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(kUseAlternativeSearchEngineProvider, false);
+  registry->RegisterIntegerPref(
+      kAlternativeSearchEngineProviderInTor,
+      TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO);
 }
 
 void InitializeSearchEngineProviderIfNeeded(Profile* profile) {
   // These search engine provider will be destroyed when observing template url
   // service is terminated.
+  // TODO(simonhong): Refactor these controller with KeyedService.
   if (profile->GetProfileType() == Profile::INCOGNITO_PROFILE) {
     new PrivateWindowSearchEngineProviderController(profile);
     return;
   }
 
+  if (profile->IsTorProfile() &&
+      profile->GetProfileType() == Profile::GUEST_PROFILE) {
```

Comment:
What's the significance of checking for GUEST_PROFILE here? Is checking for IsTorProfile not sufficient?

Reply:
yep, this is redundant unless you want to remind people tor profile is also a gust profile but that can be done in a comment. So it is better to remove this check.
IsTorProfile traits is established upon profile creation.

Reply:
This is mandatory checking if we want to use off the record profile.

Reply:
If it is tor profile, it is already a off the record profile just like guest profile. And for now this is the only way we use tor profile.